### PR TITLE
Added support to the new Audible TOC

### PR DESCRIPTION
**Pull request by nunommts**
**What's this do?**
This PR adds support to the new Audible TOC by adapting the bookmark saving and loading to the newest bookmark fields.

**Why are we doing this? (w/ JIRA link if applicable)**
There's a [feature request](https://www.notion.so/Update-Android-audiobook-player-to-support-Audible-TOC-0307c14b2f404b708d5744e6473d3187) to implement changes to the new Audible TOC on the audiobook submodule and that requires some changes on the Palace app too.

**How should this be tested? / Do these changes have associated tests?**
Follow the testing steps of [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/52)

**Dependencies for merging? Releasing to production?**
[This PR](https://github.com/ThePalaceProject/android-audiobook/pull/52) needs to be merged first. Also, [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/59) also needs to be merged first to support the latest offset changes.

**Have you updated the changelog?**
Yes

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Tested by @nunommts

### CODE REVIEW

File: simplified-bookmarks-api/src/main/java/org/nypl/simplified/bookmarks/api/BookmarkAnnotationsJSON.kt [hunk 1]
```diff
@@ -383,7 +383,8 @@ object BookmarkAnnotationsJSON {
     val objectNode = objectMapper.createObjectNode()
     objectNode.put("@type", "LocatorAudioBookTime")
     objectNode.put("chapter", bookmark.location.chapter)
-    objectNode.put("time", bookmark.location.offsetMilliseconds)
+    objectNode.put("startOffset", bookmark.location.startOffset)
+    objectNode.put("time", bookmark.location.currentOffset)
```
Comment (ray-lee) [inline simplified-bookmarks-api/src/main/java/org/nypl/simplified/bookmarks/api/BookmarkAnnotationsJSON.kt:387]: When serializing, `time` is set here to be `currentOffset`. But this doesn't seem to be what is expected by the deserialization code below. Or am I missing something?

File: simplified-bookmarks-api/src/main/java/org/nypl/simplified/bookmarks/api/BookmarkAnnotationsJSON.kt [hunk 2]
```diff
@@ -405,9 +406,12 @@ object BookmarkAnnotationsJSON {
     val obj =
       JSONParserUtilities.checkObject(null, node)
 
+    val startOffset = JSONParserUtilities.getIntegerDefault(obj, "startOffset", 0).toLong()
+
     return PlayerPosition(
       chapter = JSONParserUtilities.getIntegerDefault(obj, "chapter", 0),
-      offsetMilliseconds = JSONParserUtilities.getInteger(obj, "time").toLong(),
+      startOffset = startOffset,
+      currentOffset = JSONParserUtilities.getInteger(obj, "time").toLong() - startOffset,
```
Comment (ray-lee) [inline simplified-bookmarks-api/src/main/java/org/nypl/simplified/bookmarks/api/BookmarkAnnotationsJSON.kt:414]: When deserializing, it looks like `time` is expected to be the offset from the start of the track (which makes sense, to be backward compatible with existing bookmarks). So to get `currentOffset`, the `startOffset` value (if any) is subtracted, which also makes sense. But then, why isn't `startOffset` added on to `currentOffset` when serializing `time`?

**Review by ray-lee — COMMENTED** [review]
There seems to be a mismatch between the bookmark serialization and deserialization code. Is this a bug? I might just not be understanding it.

**Comment by nunommts** [comment]
@ray-lee I've created the audiobook PR (Also, [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/59) also needs to be merged first to support the latest offset changes.) and updated this one.

**Review by ray-lee — APPROVED** [review]
Looks good!

**Pull request merged**